### ISSUE TITLE
sql: `cockroach demo` crashes with `panic: runtime error: invalid memory address or nil pointer dereference` with TTL running

### ISSUE THREAD

**Issue by rmloveland** [cockroachdb/cockroach#77840]
Sorry if dupe, searched and did not find.

tl;dr: `cockroach demo` on 22.1 alpha-3 with some TTL tables/jobs crashed overnight on me laptop

Local system info:

```
$ uname -a
Darwin crlMBP-C02F51KPML7JMTI2.local 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_X86_64 x86_64
```

Start `cockroach demo` using the 22.1.0-alpha.3 release

```
$ cockroach demo  --insecure --global --nodes 9
# .... snip ...
# Server version: CockroachDB CCL v22.1.0-alpha.3 (x86_64-apple-darwin19, built 2022/03/14 15:31:44, go1.17.6) (same version as client)
# Cluster ID: 30fec3e0-c171-481d-8259-03c6bb6f17f8
# Organization: Cockroach Demo
```

At this point I add/drop a few tables all named `events` with TTLs, inserted some rows, etc.  (I typed a bunch of stuff - exact commands are in the attached text file: [row-level-ttl-sql-demo.txt](https://github.com/cockroachdb/cockroach/files/8254448/row-level-ttl-sql-demo.txt))

... laptop goes to sleep overnight ...

Come back in the morning and the SQL prompt shows a crash with the following backtrace:

```
root@127.0.0.1:26257/test> panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x71e9682]

goroutine 449823 [running]:
github.com/cockroachdb/cockroach/pkg/sql/ttl/ttljob.runTTLOnRange({0xc1884a0, 0xc0542c8640}, 0xc008443400, {0x76, {0x36f853a0, 0xed9c1a3d3, 0x0}}, _, {0xc0605bbef0, 0xc060b7c3f0, ...}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/ttl/ttljob/ttljob.go:628 +0x462
github.com/cockroachdb/cockroach/pkg/sql/ttl/ttljob.rowLevelTTLResumer.Resume.func2({0xc1884a0, 0xc0542c8640})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/ttl/ttljob/ttljob.go:385 +0x34d
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x92

Process cockroach exited abnormally with code 2
```

Again, see attached text file for everything I typed into the SQL prompt: [row-level-ttl-sql-demo.txt](https://github.com/cockroachdb/cockroach/files/8254448/row-level-ttl-sql-demo.txt)

~(Incidentally although it's not the point of this bug I was *not* seeing rows get deleted after my TTL timeouts - not sure if related to this or not)~ ignore, this was user error, see #77843 

Jira issue: CRDB-13793

**Comment by otan** [comment]
i'm not sure there's anything i can do here / it might be an OSX issue

The code that is segfaulting on alpha.3 is: 

https://github.com/cockroachdb/cockroach/blob/3f6fe1bb6ff498d6230a43c4307bf040f6c9d604/pkg/sql/ttl/ttljob/ttljob.go#L628

And given the `segmentation fault` (also using `--global` is suspicious as that adds crazy latency),  with the address/code being `code=0x1 addr=0x18 pc=0x71e9682`, i'm guessing this is an OSX issue

**Comment by otan** [comment]
it looks like the exact same error messages #62283 gets at random points in time albeit in a different place
going to mark this as a duplicate

**Comment by rmloveland** [comment]
This has happened again.  Same CRDB version, same `demo` startup command with same flags, same MBP laptop.

Just in case it's useful (sorry if not), sharing attached text file with all of my (very silly) interactions with  this particular cluster.  I've left it sitting for a couple days while doing other stuff and it made some long stack traces.  See attached:

[cockroach-demo-global-row-level-ttl-crash-20220323.txt](https://github.com/cockroachdb/cockroach/files/8333963/cockroach-demo-global-row-level-ttl-crash-20220323.txt)

(Actually sorry @rafiss maybe this should have been a comment on #62283 ? Anyway not sure so here it is)

**Comment by rmloveland** [comment]
Yay, thanks!  And thanks Oliver! Will see about updating to that SHA

**Comment by rafiss** [comment]
closing since the backport was completed
